psl-core/src/main/java/org/linqs/psl/config/Options.java

```diff
@@ -797,6 +797,21 @@
         + " ADAM: Update the learning rate using the Adaptive Moment Estimation (Adam) algorithm."
     );
 
+    public static final Option SGD_FIRST_ORDER_NORM = new Option(
+        "sgd.firstordernorm",
+        Float.POSITIVE_INFINITY,
+        "The p-norm used to measure the first order optimality condition."
+        + " Default is the infinity-norm which is the absolute value of the maximum component of the gradient vector.",
```

Comment:
Add in a note here that the user can represent the default value with the literal "Infinity".
(Of course they could just not supply a value and get the infinity behavior, but that would be annoying if people were scripting over multiple values.)